pkg/client/cache/watch_cache.go

```diff
@@ -27,12 +27,22 @@ import (
 	"k8s.io/kubernetes/pkg/watch"
 )
 
+// WatchCacheEvent is a single "watch event" that is send to users of
+// WatchCache. Additionally to a typical "watch.Event" it contains
+// the previous value of the object to enable proper filtering in the
+// upper layers.
```

Comment:
I'm starting to get concerned that we are building many layers now within the code that very few people will understand why they are there.  Unless there is some picture to tie it together.